resolwe_bio/tools/expressionmerge_archive.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+# pylint: disable=missing-docstring
+"""Merge columns of multiple experiments by gene id."""
+
+import argparse
+import csv
+import os
+
+import pandas as pd
+import numpy as np
+
+import utils
+
+parser = argparse.ArgumentParser(description="Merge columns of multiple experiments by gene id.")
+
+parser.add_argument('-f', '--input_file', required=True, help="Txt file with paths to expression files.")
+
+args = parser.parse_args()
+
+
+def merge_expressions(input_file):
+
+    df = pd.DataFrame(np.nan, index=[], columns=[])
+    headers = []
+
+    with open(input_file, 'r') as tfile:
+        for line in tfile:
+            line = line.rstrip()
+            header = os.path.basename(line).rstrip()
+            if header.endswith('.tab.gz'):
+                header = header[:-7]
+            headers.append(header)
+
+            with utils.gzopen(line) as csvfile:
+                reader = pd.read_csv(csvfile, index_col='Gene', delimiter='\t', dtype=str)
+                df = pd.concat([df, reader], axis=1)
+    df.columns = headers
+    return df
+
+
+def write_output(input_file, df):
+    with open(input_file, 'r') as tfile:
+        # get Species and Build from input file and use it for path of output file
+        first_line = tfile.readline().strip()
+        dir_name = os.path.split(os.path.dirname(os.path.dirname(first_line)))[0]
+        dir_name = os.path.split(dir_name)
+        # get Expression type from input file and use it in the name of output file
+        new_end_name = '_'.join([input_file.split('_')[-3], 'all_expressions.tab'])
+        name = os.path.join(dir_name[0], dir_name[1], new_end_name)
```

Comment:
If i understand correctly, `dir_name[0]` and `dir_name[1]` represent species and build, right? Please rename variables to reflect this logic.